hal/hal/rng_api.h

```diff
@@ -0,0 +1,74 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_RNG_API_H
+#define MBED_RNG_API_H
+
+#include <stddef.h>
+#include "device.h"
+
+#if DEVICE_RNG
+
+/** RNG HAL structure. rng_s is declared in the target's HAL
+ */
```

Comment:
mbed TLS has an interface for an entropy source,  whilst this is an interface for a driver for a HW RNG. Is that correct, or do we really want an interface for an entropy source?

@pjbakker suggested it makes most sense for this to remain as a TRNG API, which I think is correct, but if so, partners may not want to implement this API and instead stick with the existing API - so we'd still need both means of porting an entropy source.